src/components/NavLinks/NavLink.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC, useState } from 'react';
+import { NavLinkProps } from '@/types/props';
+import styles from './NavLink.module.scss';
+import { Link, useNavigate } from 'react-router-dom';
+import ChevronDown from '@/assets/navigation/ChevronDown';
+import Paths from '@/config/paths';
+
+const NavLink: FC<NavLinkProps> = ({ name, path, childRoutes, onLinkClicked }) => {
+  const [showSubMenu, setShowSubMenu] = useState(false);
+
+  const navigate = useNavigate();
+
+  const handleClick = (e: React.MouseEvent<HTMLElement>, path: string): void => {
+    e.preventDefault();
+
+    if (childRoutes && childRoutes?.length > 0) {
+      setShowSubMenu(!showSubMenu);
+    } else {
+      handleLinkClick(path);
+    }
+  };
+
+  const handleLinkClick = (path: string): void => {
+    if (onLinkClicked) {
+      onLinkClicked();
+    }
+
+    navigate(path);
+  };
+
+  if (!name) {
+    return;
```

Comment:
Does this not error out if returning early here? Thought TS might moan about this - I usually have to return <></> to get around that...

Reply:
now changed to `return <></>`